messaging/src/main/java/org/axonframework/eventhandling/DefaultEventProcessorSpanFactory.java

```diff
@@ -0,0 +1,199 @@
+
+        private final SpanFactory spanFactory;
+        private final boolean disableBatchTrace;
+        private final boolean distributedInSameTrace;
+        private final Duration distributedInSameTraceTimeLimit;
+
+        /**
+         * Creates a new {@link DefaultEventProcessorSpanFactory} using the provided {@code builder}.
+         *
+         * @param builder The builder to build the {@link DefaultEventProcessorSpanFactory} from.
+         */
+        protected DefaultEventProcessorSpanFactory(Builder builder) {
+            builder.validate();
+            this.spanFactory = builder.builderSpanFactory;
+            this.disableBatchTrace = builder.builderDisableBatchTrace;
+            this.distributedInSameTrace = builder.builderDistributedInSameTrace;
+            this.distributedInSameTraceTimeLimit = builder.builderDistributedInSameTraceTimeLimit;
+        }
+
+
+        /**
+         * Creates a new {@link Builder} to build a {@link DefaultEventProcessorSpanFactory} with. The default values are:
+         * <ul>
+         *     <li>{@code disableBatchTrace} defaults to {@code false}</li>
+         *     <li>{@code distributedInSameTrace} defaults to {@code false}</li>
+         *     <li>{@code distributedInSameTraceTimeLimit} defaults to 2 minutes</li>
+         * </ul>
+         * The {@code spanFactory} is a required field and should be provided.
+         *
+         * @return The {@link Builder} to build a {@link DefaultEventProcessorSpanFactory} with.
+         */
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        @Override
+        public Span createBatchSpan(boolean streaming, List<? extends EventMessage<?>> eventMessages) {
+            if (distributedInSameTrace || disableBatchTrace || !streaming) {
+                return NoOpSpanFactory.NoOpSpan.INSTANCE;
+            }
+            return spanFactory.createRootTrace(() -> "StreamingEventProcessor.batch");
+        }
+
+        @Override
+        public Span createHandleEventSpan(boolean streaming, EventMessage<?> eventMessage) {
+            if (!streaming) {
+                return spanFactory.createChildHandlerSpan(() -> "SubscribingEventProcessor.handle", eventMessage);
+            }
+            Supplier<String> name = () -> "StreamingEventProcessor.handle";
+            if (distributedInSameTrace) {
+                // Only create it in the same trace if it falls within the specified time limit.
+                if (eventMessage.getTimestamp().isAfter(Instant.now().minus(distributedInSameTraceTimeLimit))) {
+                    return spanFactory.createChildHandlerSpan(name, eventMessage);
+                }
+            }
+            if (disableBatchTrace) {
+                // No batch span, so create a linked handler span that becomes a root trace of type handler
+                return spanFactory.createLinkedHandlerSpan(name, eventMessage);
+            }
```

Comment:
This is essentially the old behavior that was covered by the TEP and PSEP through the interceptor set in the constructor, right?

Reply:
Yes, exactly! Using this flag will make it behave like it did before.

Reply:
Awesome, thanks for confirming!